tests/late_joiners.py

```diff
@@ -156,111 +130,79 @@ def run(args):
         hosts, args.binary_dir, args.debug_nodes, args.perf_nodes, pdb=args.pdb
     ) as network:
         network.start_and_join(args)
-        first_node, _ = network.find_nodes()
         all_nodes = network.get_joined_nodes()
-        term_info = find_primary(network)
-        first_msg = "Hello, world!"
-        second_msg = "Hello, world hello!"
-        catchup_msg = "Hey world!"
-        final_msg = "Goodbye, world!"
-        election_timeout = (
-            args.pbft_view_change_timeout / 1000
-            if args.consensus == "pbft"
-            else args.raft_election_timeout / 1000
+        term_info = {}
+        update_term_info(network, term_info)
+
+        test_run_txs(network=network, args=args, num_txs=TOTAL_REQUESTS)
+        update_term_info(network, term_info)
+
+        nodes_to_kill = [network.find_any_backup()]
+        nodes_to_keep = [n for n in all_nodes if n not in nodes_to_kill]
+
+        # check that a new node can catch up after all the requests
+        test_add_late_joiner(network, args, nodes_to_keep)
```

Comment:
I believe that we should prefer `network.create_and_trust_node()` here and remove `test_add_late_joiner`. `nodes_to_keep` can then be set to the difference between `network.get_joined_nodes` and `nodes_to_kill`.